staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -98,6 +98,46 @@ func ValidateUpdateCustomResourceDefinitionStatus(obj, oldObj *apiextensions.Cus
 	return allErrs
 }
 
+// ValidateCustomResourceDefinitionVersion statically validates
+func ValidateCustomResourceDefinitionVersion(version *apiextensions.CustomResourceDefinitionVersion, fldPath *field.Path, index int) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subresourcesInvalid := false
+	if version.Subresources != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceWebhookConversion"))
+			subresourcesInvalid = true
+		}
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceSubresources"))
+			subresourcesInvalid = true
+		}
+		if index == 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "subresources should not be set for first item in versions list"))
```

Comment:
you get 3 times an error if you set `Versions[0].Subresources` with the gates off.

Reply:
I think we want the error message to be as precise as possible, to tell users how they can fix the request. If `versions[1].subresources` is set with both `CustomResourceSubresources` and `CustomResourceWebhookConversion` gates off, we want to return both errors.

For `versions[0].subresources` arguably we don't need to check the feature gates if we disallow setting `versions[0]` fields. We should tell users to fix the request by having `versions[0].subresources = nil`. Updated the validation code so that we do not check feature gates for `versions[0]`.

After another thought, @sttts do you suggest we should aggregate the detail messages for the same path into a single Error?